doc/secure_upgrade/secure_upgrade.md

```diff
@@ -0,0 +1,179 @@
+		* 1.9.1. [CLI/YANG model Enhancements](#CLIYANGmodelEnhancements)
+		* 1.9.2. [Config DB Enhancements](#ConfigDBEnhancements)
+	* 1.10. [Warmboot and Fastboot Design Impact](#WarmbootandFastbootDesignImpact)
+	* 1.11. [ Restrictions/Limitations](#RestrictionsLimitations)
+		* 1.11.1. [ uefi tools and secure boot enabled](#uefitoolsandsecurebootenabled)
+		* 1.11.2. [ ONIE secure upgrade verification code](#ONIEsecureupgradeverificationcode)
+	* 1.12. [ Upgrade and downgrade flows](#Upgradeanddowngradeflows)
+		* 1.12.1. [ secure upgrade from non secured SONIC/ONIE to secured SONIC](#secureupgradefromnonsecuredSONICONIEtosecuredSONIC)
+		* 1.12.2. [ secure upgrade from secured SONIC to secured SONIC](#secureupgradefromsecuredSONICtosecuredSONIC)
+		* 1.12.3. [ downgrade from secured SONIC to non-secured SONIC](#downgradefromsecuredSONICtonon-securedSONIC)
+		* 1.12.4. [secure upgrade from secure upgrade enabled ONIE to secured SONIC](#secureupgradefromsecureupgradeenabledONIEtosecuredSONIC)
+	* 1.13. [ Testing Requirements/Design](#TestingRequirementsDesign)
+		* 1.13.1. [Unit Test cases](#UnitTestcases)
+		* 1.13.2. [System Test cases](#SystemTestcases)
+	* 1.14. [Open/Action items - if any](#OpenActionitems-ifany)
+
+<!-- vscode-markdown-toc-config
+	numbering=true
+	autoSave=true
+	/vscode-markdown-toc-config -->
+<!-- /vscode-markdown-toc -->
+
+###  1.1. <a name='Revision'></a>Revision
+  Rev  |  Date   |      Author      | Change Description |
+| :---: | :-----: | :--------------: | ------------------ |
+|  0.1  | 6/2022 |   Yona Cohen   | Phase 1 Design     |
+
+###  1.2. <a name='Scope'></a>Scope
+
+This secure upgrade hld doc described the requirements, architecture, and general flow details of secure upgrade feature in switches SONIC OS based.
+
+###  1.3. <a name='DefinitionsAbbreviations'></a>Definitions/Abbreviations
+
+	SU - secure upgrade
+	SB - secure boot	
+	CMS - Cryptographic Message Syntax
+
+###  1.4. <a name='Overview'></a>Overview
+
+Secure installation and upgrade of OS are setting a requirement to ensure the image was not modified since it was built by the vendor. This is a basic requirement as part of general OS security. In order to help the credibility of the OS image, secure upgrade mechanism guarantees the image was not modified by any unauthorized entity since its original creation.
+
+###  1.5. <a name='Requirements'></a>Requirements
+
+We want to enable secure upgrade of SONIC. This will include secure upgrade of SONIC while installing SONIC via ONIE or SONIC. The feature requires a signing process which will include a private key and certificate. The verification process on the other end requires a public key certificate for validation. Both processes are required to be supported by various scenarios such as development and release.
+In order to do secure installation of SONIC from ONIE, it has to include the flowing secure upgrade verification commit TBD (on pull request - waiting for it to be created)
```

Comment:
correct typo "flowing" here - is the secure upgrade verification commit ready? if yes please share it

Reply:
fix the typo, and for the moment the "commit" is waiting for review, added the current PR in the HLD